module/skill/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,382 @@
+            const float dt = get_time_delta();
+            current_orders.setZero();
+            walk_engine.update_state(dt, current_orders);
+            imu_reaction.enable(false);
+        });
+
+        // MAIN LOOP
+        on<Provide<Walk>,
+           Needs<LeftLegIK>,
+           Needs<RightLegIK>,
+           Every<UPDATE_FREQUENCY, Per<std::chrono::seconds>>,
+           Single>()
+            .then([this](const Walk& walk) {
+                // TODO: Set this based on the walk engine state
+                if (walk_engine.get_state() == WalkEngineState::IDLE) {
+                    emit(std::make_unique<Stability>(Stability::STANDING));
+                }
+                else {
+                    emit(std::make_unique<Stability>(Stability::DYNAMIC));
+                }
+                // the engine expects orders in [m] not [m/s]. We have to compute by dividing by step frequency which is
+                // a double step factor 2 since the order distance is only for a single step, not double step
+                const float factor             = (1.0f / (current_cfg.params.freq)) * 0.5f;
+                const Eigen::Vector3f& command = walk.velocity_target.cast<float>() * factor;
+
+                // Clamp velocity command
+                current_orders =
+                    command.array().max(-current_cfg.max_step.array()).min(current_cfg.max_step.array()).matrix();
+
+                // translational orders (x+y) should not exceed combined limit. scale if necessary
+                if (current_cfg.max_step_xy != 0) {
+                    float scaling_factor =
+                        1.0f / std::max(1.0f, (current_orders.x() + current_orders.y()) / current_cfg.max_step_xy);
+                    current_orders.cwiseProduct(Eigen::Vector3f(scaling_factor, scaling_factor, 1.0f));
+                }
+
+                // warn user that speed was limited
+                if (command.x() != current_orders.x() || command.y() != current_orders.y()
+                    || command.z() != current_orders.z()) {
+                    log<NUClear::WARN>(fmt::format(
+                        "Speed command was x: {} y: {} z: {} xy: {} but maximum is x: {} y: {} z: {} xy: {}",
+                        command.x(),
+                        command.y(),
+                        command.z(),
+                        command.x() + command.y(),
+                        current_cfg.max_step.x() / factor,
+                        current_cfg.max_step.y() / factor,
+                        current_cfg.max_step.z() / factor,
+                        current_cfg.max_step_xy / factor));
+                }
+
+                const float dt = get_time_delta();
+                // If the walk engine has new goals, call the calculate_joint_goals function,
+                // which will emit the tasks for the limbs.
+                // If there are no new goals, no tasks are emitted, removing child tasks from the director tree.
```

Comment:
This comment is kind of incorrect. It's not if there's no new goals, it's if the walk engine is stopped/not moving currently. The update state function will return false if the walk isn't running/idle/paused and will return true if it's moving. 
If it was for new goals, we wouldn't be destroying the subtasks (because that would stop the walk), we'd want to Idle so it keeps doing the last thing.